pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -0,0 +1,362 @@
+}
+
+func (srv AlertmanagerSrv) RouteDeleteSilence(c *models.ReqContext) response.Response {
+	silenceID := c.Params(":SilenceId")
+	srv.log.Info("RouteGetSilence: ", "SilenceId", silenceID)
+	q := ngmodels.DeleteSilenceByUIDCommand{OrgID: c.SignedInUser.OrgId, UID: silenceID}
+	if err := srv.store.DeleteSilenceByUID(&q); err != nil {
+		if errors.Is(err, ngmodels.ErrSilenceNotFound) {
+			return response.Error(http.StatusNotFound, "silence not found", err)
+		}
+		return response.Error(http.StatusInternalServerError, "failed to delete silence", err)
+	}
+	return response.JSON(http.StatusOK, util.DynMap{"message": "silence deleted"})
+}
+
+func (srv AlertmanagerSrv) RouteGetAlertingConfig(c *models.ReqContext) response.Response {
+	query := ngmodels.GetLatestAlertmanagerConfigurationQuery{}
+	err := srv.store.GetLatestAlertmanagerConfiguration(&query)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to get latest configuration", err)
+	}
+
+	cfg := apimodels.PostableUserConfig{}
+	err = yaml.Unmarshal([]byte(query.Result.AlertmanagerConfiguration), &cfg)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to unmarshal alertmanager configuration", err)
+	}
+
+	var apiReceiverName string
+	var receivers []*apimodels.GettableGrafanaReceiver
+	alertmanagerCfg := cfg.AlertmanagerConfig
+	if len(alertmanagerCfg.Receivers) > 0 {
+		apiReceiverName = alertmanagerCfg.Receivers[0].Name
+		receivers = make([]*apimodels.GettableGrafanaReceiver, 0, len(alertmanagerCfg.Receivers[0].PostableGrafanaReceivers.GrafanaManagedReceivers))
+		for _, pr := range alertmanagerCfg.Receivers[0].PostableGrafanaReceivers.GrafanaManagedReceivers {
+			secureFields := make(map[string]bool, len(pr.SecureSettings))
+			for k := range pr.SecureSettings {
+				secureFields[k] = true
+			}
+			gr := apimodels.GettableGrafanaReceiver{
+				Uid:                   pr.Uid,
+				Name:                  pr.Name,
+				Type:                  pr.Type,
+				IsDefault:             pr.IsDefault,
+				SendReminder:          pr.SendReminder,
+				DisableResolveMessage: pr.DisableResolveMessage,
+				Frequency:             pr.Frequency,
+				Settings:              pr.Settings,
+				SecureFields:          secureFields,
+			}
+			receivers = append(receivers, &gr)
+		}
+	}
+
+	gettableApiReceiver := apimodels.GettableApiReceiver{
+		GettableGrafanaReceivers: apimodels.GettableGrafanaReceivers{
+			GrafanaManagedReceivers: receivers,
+		},
+	}
+	gettableApiReceiver.Name = apiReceiverName
+	result := apimodels.GettableUserConfig{
+		TemplateFiles: cfg.TemplateFiles,
+		AlertmanagerConfig: apimodels.GettableApiAlertingConfig{
+			Config: alertmanagerCfg.Config,
+			Receivers: []*apimodels.GettableApiReceiver{
+				&gettableApiReceiver,
+			},
+		},
+	}
+
+	return response.JSON(http.StatusOK, result)
+}
+
+func (srv AlertmanagerSrv) RouteGetAmAlertGroups(c *models.ReqContext) response.Response {
```

Comment:
This nit again :)
```suggestion
func (srv AlertmanagerSrv) RouteGetAMAlertGroups(c *models.ReqContext) response.Response {
```

Reply:
yep, I need to sync this branch with the other one.